### ISSUE TITLE
Magisk 19.4+ cause instabilities.

### ISSUE THREAD

**Issue by asislife20** [Magisk-Modules-Repo/com.piyushgarg.rclone#31]
I just recently updated magisk to 19.4, and having rclone installed while on 19.4 beta or above canary builds cause my device to freeze, and then reboot. As soon as I uninstalled rclone, device started functioning as normal. I am on Moto z3 Play Android 9.0. I'll be going back to 19.3 because I love rclone, but thought I'd let you know about the issues before 19.4 stable is released.

**Comment by piyushgarg** [comment]
Thanks @asislife for reporting the issue early and trying it rclone on Android. I suggest you to wait for the stable release of 19.4. I will check if there is indeed a problem.
Also let me know the steps to go to previous version of magisk.

**Comment by AleD219** [comment]
Same here, with 19.5 canary
Devices freezes, shows a black screen and then, reboots

**Comment by HeyItsJono** [comment]
Magisk 19.4 (stable), also getting device freezes. In my case, the device locks up and hangs entirely maybe a minute after booting when this module is enabled, and requires a forced reset via holding down the Power + VolDown keys to turn it off. When the module is disabled this no longer occurs. My setup is just a basic rclone config file with a GDrive in it, no fancy advanced/custom params or anything. Magisk 19.4 is stable now, no longer a canary or beta build.

**Comment by asislife20** [comment]
As far as I'm aware 19.4 is still in beta, not stable yet.

**Comment by HeyItsJono** [comment]
Apologies, you're right, I accidentally had the manager set to Beta channel. Does Magisk 19.3 have Android 10 support, or is it just 19.4+?

**Comment by SJ50** [comment]
even rclone config command also leads to instability. 
I can not downgrade to 19.3, so I have to wait until this issue has been resolved.

**Comment by asislife20** [comment]
Why can't you downgrade?? to 19.3?

**Comment by SJ50** [comment]
@asislife20 19.4 has require changes to root my Nvidia Shield TV.

**Comment by SJ50** [comment]
@piyushgarg @Geofferey I was able to run rclone by changing 

> UPDDIR=/data/adb/modules_update

to 

> UPDDIR=/data/adb/modules

no crashing but all the mount points are empty

**Comment by SJ50** [comment]
I did some experiments, first remote does not get mounted at all but second one does and as soon as I add .param file global or specific to remote all remote fails to mount.

**Comment by gulp79** [comment]
Is this rclone module working ok with current v20 magisk release?

**Comment by asislife20** [comment]
Don't want to speak to soon, but it seems as though magisk V20 works with rclone without issues, but I've only been on V20 for about 20 mins. Can someone else please confirm if it fixed the issues for you?

**Comment by SJ50** [comment]
After reboot system hangs on startup and requires reboot.

**Comment by asislife20** [comment]
I spoke to soon, did the same for me too

**Comment by gulp79** [comment]
Yes this mudule is unusable and hangs the phone on magisk >19.3. I've just updated my phone to android Q, and I've to use magisk 20 or I've a bootloop. So atm I've to disable this module and hope for a fix.

**Comment by f1r3w411** [comment]
Module works correctly for me with the Magisk v19.3.
Nevertheless, it doesn't work with the latest Magisk v20.0 stable.
Have anyone already checked this issue?

**Comment by HASJ** [comment]
Yep. Broken on v20.

**Comment by RaXorX** [comment]
I'm not even able to run config or set it up, on PE beta Q with magisk 20. Running rclone config results to nothing and after a while, phone stops working.

**Comment by SJ50** [comment]
I am on V20. and I made following changes on all relevant files.

UPDDIR=/data/adb/modules_update

to

UPDDIR=/data/adb/modules

No crashing everything works smoothly. Tested for 3 days with multiple reboot.

**Comment by asislife20** [comment]
Anyway I could get a copy of the modified files?

**Comment by gulp79** [comment]
Waiting for an official release you can use this:
https://mega.nz/#!J1R1ECLK

**Comment by asislife20** [comment]
Mega link needs a decryption key

**Comment by gulp79** [comment]
Sorry.. Updated

**Comment by asislife20** [comment]
Works perfect, thank you! @gulp79

**Comment by HeyItsJono** [comment]
@gulp79 Confirmed this doesn't crash my phone, but where does it mount to? There are mounts under `/mnt/cloud/` or `/storage/cloud`

**Comment by SJ50** [comment]
@HeyItsJono it should be /mnt/cloud

**Comment by asislife20** [comment]
Magisk update 20.1 seems to have fixed the issues for me!

**Comment by piyushgarg** [comment]
directly upgraded to 20.1 and it seems to be working fine.

**Comment by thekiefs** [comment]
@piyushgarg these issues are back on the latest Canary version of Magisk

**Comment by thekiefs** [comment]
Can confirm that @gulp79 build fixes it!

**Comment by thekiefs** [comment]
@gulp79 I just installed Magisk 20.4 and Manager 7.5.1. I installed your build. It works great immediately after installing. I can access my remote at /storage/cloud/googledrive, etc. However as soon as I reboot my mount disappears and I have to reinstall your build again to get /storage/cloud/googledrive again. Any help would be appreciated! Maybe @piyushgarg @Geofferey can update the official module as well.
